app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,119 @@
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    private var step: Int = 1
+    private var unit: String = ""
+
+    @Suppress("unused")
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) :
+            this(context, attrs, R.attr.preferenceStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        onPreferenceChangeListener = OnPreferenceChangeListener { _, newValue ->
+            summary = getTextForValue(newValue.toString())
+            true
+        }
+        onPreferenceClickListener = OnPreferenceClickListener {
+            showSeekBarDialog()
+            true
+        }
+    }
+
+    override fun onAttachedToHierarchy(preferenceManager: PreferenceManager?) {
+        super.onAttachedToHierarchy(preferenceManager)
+        summary = getTextForValue(
+            preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+        )
+    }
+
+    /**
+     * Generates the text for the given [value] and adds the defined [unit] at the end.
+     */
+    private fun getTextForValue(value: Any): String {
+        return if (value !is Int) {
+            "??$unit"
+        } else {
+            value.toString() + unit
+        }
+    }
+
+    /**
+     * Shows the seek bar dialog.
+     */
+    private fun showSeekBarDialog() {
+        val dialogView = DialogSeekBarBinding.inflate(LayoutInflater.from(context))
+        val initValue = preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+        dialogView.seekBar.max = actualValueToSeekBarProgress(max)
+        dialogView.seekBar.progress = actualValueToSeekBarProgress(initValue)
+        dialogView.seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                dialogView.seekBarValue.text = getTextForValue(seekBarProgressToActualValue(progress))
+            }
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {}
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {}
+        })
+        dialogView.seekBarValue.text = getTextForValue(initValue)
+        AlertDialog.Builder(context).apply {
+            setTitle(this@DialogSeekBarPreference.title)
+            setCancelable(true)
+            setView(dialogView.root)
+            setPositiveButton(android.R.string.ok) { _, _ ->
+                val actualValue = seekBarProgressToActualValue(dialogView.seekBar.progress)
+                preferenceDataStore?.putInt(key, actualValue)
+            }
+            setNegativeButton(android.R.string.cancel, null)
+            setOnDismissListener { summary = getTextForValue(
+                preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
```

Comment:
Consider extract this line as a method or getter? It's repeated several times in this file.